public/components/trace_analytics/components/common/plots/__tests__/service_map.test.tsx

```diff
@@ -3,25 +3,243 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { configure, shallow } from 'enzyme';
+import { configure } from 'enzyme';
 import Adapter from 'enzyme-adapter-react-16';
+import { mount } from 'enzyme';
 import React from 'react';
-import { TEST_SERVICE_MAP } from '../../../../../../../test/constants';
+import { act } from '@testing-library/react';
 import { ServiceMap } from '../service_map';
+import { EuiFieldSearch, EuiSelectable } from '@elastic/eui';
+import { TEST_SERVICE_MAP } from '../../../../../../../test/constants';
+import Graph from 'react-graph-vis';
+
+configure({ adapter: new Adapter() });
+
+// Mock uuid
+jest.mock('uuid', () => ({
+  v4: jest.fn(() => 'static-uuid'),
+}));
+
+// Normalize dynamic values in snapshots
+expect.addSnapshotSerializer({
+  test: (val) => typeof val === 'string' && /^[a-f0-9-]{36}$/.test(val),
+  print: () => '"<dynamic-uuid>"',
+});
+
+// Mock crypto.getRandomValues
+const crypto = {
+  getRandomValues: jest.fn((arr) => arr.fill(0)), // Fill with consistent values
+};
+Object.defineProperty(global, 'crypto', { value: crypto });
+
+const mockContext = ({
+  canvas: document.createElement('canvas'),
+  fillRect: jest.fn(),
+  clearRect: jest.fn(),
```

Comment:
bigger mocks can be moved to `test/contants`

Reply:
Moved to MOCK_CANVAS_CONTEXT in constants file